packages/core/src/indexing/build-indexing.ts

```diff
@@ -45,11 +46,13 @@ function parseURL(input: string) {
  * Build DatabaseIndexAPI instance based on passed indexing configuration.
  */
 export function buildIndexing(indexingConfig: IndexingConfig): DatabaseIndexApi {
+  const diagnosticsLogger = new LoggerProvider().getDiagnosticsLogger()
```

Comment:
IMO, better to pass the `diagnosticsLogger` from a caller.

Reply:
good catch, added it since it was already available from within the caller.